### PR TITLE
Fix onClosing() web socket handler in OKHttp client (#1262).

### PR DESCRIPTION
**Pull request by dmitrievanthony**
**Subsystem**
OkHttp client.

**Motivation**
The initial problem is mentioned in https://github.com/ktorio/ktor/issues/1262.

**Solution**
Add `onClosing()` handler to `OkHttpWebsocketSession.kt`.

### CODE REVIEW

File: ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt [hunk 1]
```diff
@@ -93,6 +93,14 @@ internal class OkHttpWebsocketSession(
             "${CloseReason.Codes.byCode(code.toShort())?.toString() ?: code}."))
     }
 
+    override fun onClosing(webSocket: WebSocket, code: Int, reason: String) {
+        super.onClosing(webSocket, code, reason)
+
+        _closeReason.complete(CloseReason(code.toShort(), reason))
+        outgoing.sendBlocking(Frame.Close(CloseReason(code.toShort(), reason)))
```
Comment (cy6erGn0m) [inline ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt]: Sending to a closed channel leads to an exception. Not sure what could happen when throwing it to okhttp

**Comment by cy6erGn0m** [comment]
Looks like the test doesn't pass on CIO :)

**Comment by dmitrievanthony** [comment]
> Looks like the test doesn't pass on CIO :)

Yes, looks like CIO has the same problem. I'm figuring out.

File: ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt (hunk 1 above)
Comment (dmitrievanthony) [reply ktor-client/ktor-client-okhttp/jvm/src/io/ktor/client/engine/okhttp/OkHttpWebsocketSession.kt]: I added an additional check here, but still not sure about what we shell do in such a case.

**Pull request merged**